src/Printers/Printer.ts

```diff
@@ -0,0 +1,374 @@
+import { IDocument } from '../Documents/Commands';
+import { ConfigDocumentBuilder, IConfigDocumentBuilder } from '../Documents/ConfigDocument';
+import {
+    ILabelDocumentBuilder,
+    LabelDocumentBuilder,
+    LabelDocumentType
+} from '../Documents/LabelDocument';
+import { NumericRange } from '../NumericRange';
+import { WebZlpError } from '../WebZlpError';
+import {
+    IPrinterDeviceChannel,
+    PrinterChannelType,
+    PrinterCommunicationOptions
+} from './Communication/PrinterCommunication';
+import { UsbPrinterDeviceChannel } from './Communication/UsbPrinterDeviceChannel';
+import { PrinterCommandLanguage, PrinterOptions, PrintSpeed } from './Configuration/PrinterOptions';
+import { EplPrinterCommandSet } from './Languages/EplPrinterCommandSet';
+import { PrinterCommandSet } from './Languages/PrinterCommandSet';
+import { ZplPrinterCommandSet } from './Languages/ZplPrinterCommandSet';
+import { PrinterModelDb } from './Models/PrinterModel';
+
+/**
+ * A class for working with a label printer.
+ */
+export class Printer {
```

Comment:
You need unit tests for all these files! :P

Reply:
Goodness do I.